ansible/TestbedProcessing.py

```diff
@@ -416,6 +418,89 @@ def makeLab(data, devices, testbed, outfile):
                                 entry += "\tansible_ssh_pass=" + ansible_ssh_pass
                             except:
                                 print("\t\t" + host + ": ansible_ssh_pass not found")
+                        try: #get hwsku
+                            hwsku = dev.get("hwsku")
+                            if hwsku is not None:
+                               entry += "\thwsku=" + hwsku
+                        except:
+                            print("\t\t" + host + ": hwsku not found")
+
+                        try: #get cardtype
+                            cardtype = dev.get("cardtype")
+                            if cardtype is not None:
+                               entry += "\tcardtype=" + cardtype
+                        except:
+                           print("\t\t" + host + ": cardtype not found")
+
+                        try: #get num_asics
+                            num_asic = dev.get("num_asic")
+                            if num_asic is not None:
+                               entry += "\tnum_asic=" + str( num_asic )
```

Comment:
I believe the variable name in the inventory/lab file should be 'num_asics', and not 'num_asic'

Reply:
the variable num_asics is already used in config_sonic_basedon_testbed.yml.
I can rename the new num_asic to num_asics if there is no conflict in my diffs with existing assumptions of num_asics

Reply:
@SuvarnaMeenakshi should there be num_asic and also num_asics ? 
In most of the places it uses {{ num_asic }} and in one place it uses {{ num_asics }} .
Why do we need both ?   should it be just one ? 

ansible/config_sonic_basedon_testbed.yml:    port_alias: hwsku="{{ hwsku }}" num_asic="{{ num_asics }}"
ansible/roles/vm_set/tasks/kickstart_vm.yml:      num_asic: "{{ hostvars[vm_name]['num_asics'] | default(1) }}"
ansible/roles/vm_set/tasks/kickstart_vm.yml:              num_asic={{ num_asic }}
ansible/roles/vm_set/tasks/kickstart_vm.yml:              num_asic={{ num_asic }}
ansible/roles/vm_set/tasks/start_sonic_vm.yml:    num_asic: "{{ hostvars[dut_name]['num_asics'] | default(1) }}"
ansible/roles/vm_set/tasks/start_sonic_vm.yml:  port_alias: hwsku={{ hostvars[dut_name].hwsku }} num_asic={{ num_asic }}
ansible/roles/vm_set/tasks/start_sonic_vm.yml:             num_asic={{ num_asic }}
ansible/library/port_alias.py:      port_alias: hwsku='ACS-MSN2700' num_asic=1
ansible/library/port_alias.py:            num_asic=dict(type='int', required=False)

Reply:
ansible/roles/vm_set/tasks/kickstart_vm.yml uses {{ num_asic }} and not {{ num_asics }}
where is the num_asic coming from ?

Reply:
> ansible/roles/vm_set/tasks/kickstart_vm.yml uses {{ num_asic }} and not {{ num_asics }}
> where is the num_asic coming from ?

https://github.com/Azure/sonic-mgmt/blob/master/ansible/roles/vm_set/tasks/kickstart_vm.yml#L53 - Here a variable "num_asic" is created which gets the data from "num_asics" in inventory file.
```
num_asic: "{{ hostvars[vm_name]['num_asics'] | default(1) }}"
```
The variable num_asic is referenced every where else in the file.
Ex: https://github.com/Azure/sonic-mgmt/blob/master/ansible/roles/vm_set/tasks/kickstart_vm.yml#L63 - here num_asic variable defined in line 53 above is used.

Reply:
num_asics and num_asic are very close and leads to confusion. 
should num_asic be renamed to num_asic_vm in kickstart_vm.yml and num_asic_dut in start_sonic_vm.yml ? or just use num_asics everywhere ?

Reply:
> num_asics and num_asic are very close and leads to confusion.
> should num_asic be renamed to num_asic_vm in kickstart_vm.yml and num_asic_dut in start_sonic_vm.yml ? or just use num_asics everywhere ?

could use num_asics everywhere if that makes it easier, as the variable is locally used.

Reply:
I can rename all the num_asic to num_asics part of https://github.com/Azure/sonic-mgmt/pull/3245 or should that be done in a separate PR ?

Reply:
I suggest having separate PR.